### DISCUSSION TITLE
Removing `@auth` directive seems to have no effect

### DISCUSSION THREAD

**Discussion by mrcleanandfresh** [accounts-js/accounts#1238] [Q&A] [answered]
I'm using Fastify as a server, instead of Express; and [Mercurius](https://github.com/mercurius-js/mercurius) as my GraphQL layer instead of Apollo. The schema building code in question looks like this:
```typescript
const accountsGraphQLPlugin: FastifyPluginCallback<AccountsGraphQLPluginOptions> =
  (fastify, opts, done) => {
    // set up the Accounts GraphQL server.
    const accountsGraphQL = AccountsModule.forRoot({
      accountsServer: fastify.accounts,
    });

    fastify.decorate('accountsGraphQL', accountsGraphQL);

    const schema = makeExecutableSchema({
      typeDefs: mergeTypeDefs([accountsGraphQL.typeDefs, typeDefs]),
      resolvers: mergeResolvers([accountsGraphQL.resolvers, resolvers]),
      schemaDirectives: {
        ...accountsGraphQL.schemaDirectives, // <---- installs the `@auth` directive into my schema
      },
    });
   //... truncated
}
```
I have extended the schema provided by the `AccountsModule`, and it looks like this:
```graphql
  extend type Mutation {
    #... truncated
    sendResetPasswordEmail(email: String!): Boolean @auth
    #... truncated
  }
```
I am using the [authentication directive](https://github.com/accounts-js/accounts/tree/master/packages/graphql-api/src/utils) provided by `@accounts/graphql-api` on the `sendResetPasswordEmail`, but when I remove the directive:

```graphql
  extend type Mutation {
    #... truncated
    sendResetPasswordEmail(email: String!): Boolean
    #... truncated
  }
```
I am still getting [this error message](https://github.com/accounts-js/accounts/blob/master/packages/graphql-api/src/utils/authenticated-resolver.ts#L21). I have no idea why, but I dug a little deeper anyway.

I put a breakpoint on `makeExecutableSchema` and a breakpoint in [`authenticated-directive.ts`](https://github.com/accounts-js/accounts/blob/master/packages/graphql-api/src/utils/authenticated-directive.ts#L6). Once `makeExecutableSchema` is called (before any of my custom code is executed) the directive field gets called with `sendResetPasswordEmail`. I'm using [`mercurius-auth`](https://github.com/mercurius-js/auth) and thought that was the problem, but don't think so for two reasons 1 - it binds to the directive well-before mercurius-auth is initialized and 2 - when I commented it out I saw the same behavior.

This points to it either being a problem with my schema (shown above), [the `AccountsModule` schema](https://github.com/accounts-js/accounts/blob/master/packages/graphql-api/src/modules/accounts-password/schema/mutation.ts#L11), my resolver (has no HOF for authentication) or [`AccountsModule` resolver](https://github.com/accounts-js/accounts/blob/master/packages/graphql-api/src/modules/accounts-password/resolvers/mutation.ts#L100-L120). I've looked in all these places and haven't seen anything obvious.

So I have no idea what's going on, I'm pretty stuck. Have any of your seen this before? I am using Mercurius Codegen (like GraphQL Codegen) but it's called [`mecurius-typescript`](https://github.com/mercurius-js/mercurius-typescript) but that only generates TypeScript and couldn't be caching something that would affect the directive. I get the same error whether I'm using GraphiQL, or Postman. I'm stumped.

## Update 9-17-2022

When I commented out the schema directives:

```typescript
    const schema = makeExecutableSchema({
      typeDefs: mergeTypeDefs([accountsGraphQL.typeDefs, typeDefs]),
      resolvers: mergeResolvers([accountsGraphQL.resolvers, resolvers]),
      schemaDirectives: {
        // ...accountsGraphQL.schemaDirectives, // <---- installs the `@auth` directive into my schema
      },
    });
```

This uses my existing authentication. The request went through just fine. So probably something about how the `accountsGraphQL.schemaDirectives` is registered, or applied to the `typeDefs`?

I'm going to use this as a workaround for now, since I'm using mercurius auth instead, and it's more advanced than just testing for a null user.

**Comment by darkbasic** [comment, marked as answer]
Should be fixed in 1.0: https://github.com/accounts-js/accounts/pull/1258
